_posts/blog/2024-08-30-accumulo4-preview.md

```diff
@@ -0,0 +1,73 @@
+## New For 4.0
+
+The features in version 4.0 are intended to make running Accumulo in a cloud environment more cost efficient by building on the optional and experimental features added in version 2.1. Prior to version 4.0 running an Accumulo instance required enough compute resources to host enough TabletServers to support the ingest, query, and Tablet maintenance (compact, split, merge, etc.) workload as Accumulo was originally designed to keep all Tablets immediately accessible all the time. Version 4.0 allows the user more flexibility in how they deploy Accumulo server processes and how they interact with their data. Below we introduce the high-level features/changes included in version 4.0.
+
+### On-Demand Tablets
+
+On an upgrade to Accumulo 4.0 the upgrade code will assign all Tablets, except for the root and metadata tables, with an availability setting of ONDEMAND. What this means is that the Tablet is not assigned and hosted by default. If an operation is performed that requires a Tablet to be hosted by a TabletServer, then the operation will wait for the Tablet to be assigned and hosted. This setting can be changed and checked using the Shell commands setavailability and getavailability, respectively. When a configurable amount of time has passed where the Tablet has been unused, then it will be unloaded. Other valid availability values are HOSTED, which means that the Tablet will always be hosted (the default in earlier versions of Accumulo), and UNHOSTED, which means that the Tablet will never be hosted. 
+
+User operations that would require a Tablet to be hosted are live ingest and immediate consistency scans. Users can still interact with data in unhosted tablets via bulk import and eventual consistency scans, and users can still perform tablet maintenance operations on unhosted tablets. The root and metadata tables have an availability value of HOSTED, which cannot be changed by the user. If your application only performs eventual scans and bulk imports, then only one TabletServer is required to host the root and metadata tables.
+
+Because Tablets are now optionally hosted in a TabletServer, the implementation of all the Tablet maintenance functions had to be moved out of the TabletServer and re-implemented. Split, Merge, and other metadata-only operations were re-implemented as Fate operations in the Manager.
+
+### External Compactions Only
+
+If a Tablet is not hosted, and the user is bulk importing to it, this could trigger the need for a major compaction. Hosting the Tablet just for the purpose of compacting it will cause churn on the cluster as the balancer may move Tablets around. This led to the decision to move all major compactions to the ExternalCompactions feature. In 4.0 the CompactionCoordinator component was merged into the Manager process, so running the CompactionCoordinator process is no longer required. Running at least one Compactor is required to perform major compactions on the root and metadata tables.
```

Comment:
"in 4.0"
-->
"in 4.0,"

"In 4.0 the CompactionCoordinator component was merged into the Manager process, so running the CompactionCoordinator process is no longer required."
-->
"In 4.0 the CompactionCoordinator component was merged into the Manager process, so **manually** running the CompactionCoordinator process is no longer required."

Reply:
Addressed in 40c6217